src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -1319,41 +1319,60 @@ impl ToStream for LogicalJoin {
         let r2i = join.core.r2i_col_mapping().composite(&mapping);
 
         // Add missing pk indices to the logical join
-        let left_to_add = left
+        let mut left_to_add = left
             .logical_pk()
             .iter()
             .cloned()
-            .filter(|i| l2i.try_map(*i).is_none());
+            .filter(|i| l2i.try_map(*i).is_none())
+            .collect_vec();
 
-        let right_to_add = right
+        let mut right_to_add = right
             .logical_pk()
             .iter()
             .cloned()
             .filter(|i| r2i.try_map(*i).is_none())
```

Comment:
Similar here, I think we actually want r2o?

Reply:
Sure.